### ISSUE TITLE
$this in closure is not traversable

### ISSUE THREAD

**Issue by TomBZombie** [scrutinizer-ci/php-analyzer#612]
> Here, you can explain why the found issue was invalid, and what result
> you would have expected otherwise.
>
> Please also check that this issue has not already been reported.

see https://scrutinizer-ci.com/g/TomBZombie/Maphper/code-structure/master/class/Maphper%5CDataSource%5CDatabase?issueId=5b35d617-4cac-4063-9fed-07a329f1210a

This is showing that this code is invalid:

```php
		$readClosure = function() {
			$data = new \stdClass;
			foreach ($this as $k => $v)	$data->$k = $v;
			return $data;
		};
		$read = $readClosure->bindTo($data, $data);
		$writeData = $read();
```

While it's possible that the $this reference is not referencing something that is traversable,  there's no way you can possibly know that using static analysis because ` $readClosure->bindTo($data, $data);` binds the closure to another object. 

Even ignoring that, however, objects in PHP are almost always traversable. `foreach ($this )` will iterate over the properties on the object if it doesn't implement traversable.

**Comment by jcheron** [comment]
I agree with that : php objects are always traversable, see http://php.net/manual/en/language.oop5.iterations.php

**Comment by elliotchance** [comment]
http://phpepl.herokuapp.com?code=%24a%20%3D%20json_decode('%7B%22a%22%3A%22b%22%2C%20%22c%22%3A%22d%22%7D')%3B%0Avar_dump(%24a)%3B%0Aforeach%20(%24a%20as%20%24k%20%3D%3E%20%24v)%20%7B%0A%20%20echo%20%22%24k%20%24v%5Cn%22%3B%0A%7D%0A
